### PR TITLE
Upgrade Drupal core to 8.5.0, plus other outdated modules.

### PR DESCRIPTION
**Pull request by ruscoe**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -40,31 +40,33 @@
         "drupal/config_split": "^1.1",
         "drupal/console": "~1.0",
         "drupal/contact_storage": "1.0.0-beta8",
-        "drupal/core": "8.4.*",
+        "drupal/core": "8.5.0-beta1",
```
Comment (unclegcb) [inline composer.json]: Hm. I guess we should probably start all new projects on this Beta, if we start them soon, but I also want to make sure we don't get stuck at the Beta. Does `8.5.x` not work here?

File: composer.json [hunk 2]
```diff
@@ -40,31 +40,33 @@
...
         "drupal/redirect": "^1.0-alpha5",
         "drupal/redis": "^1.0-beta1",
-        "drupal/simple_sitemap": "2.10",
+        "drupal/simple_sitemap": "2.11",
         "drupal/smart_trim": "^1.0",
-        "drupal/token": "1.0",
+        "drupal/token": "1.1",
         "drush/drush": "^8.1.9",
+        "symfony/config": "3.4.x-dev",
+        "symfony/dependency-injection": "3.4.x-dev",
```
Comment (unclegcb) [inline composer.json]: tying to Dev feels dangerous. A requirement for 8.5 beta?

**Review by unclegcb — COMMENTED** [review]
Couple questions inline if you have thought @ruscoe

**Comment by ruscoe** [comment]
@unclegcb Removed references to dev branches, now using last stable release for symfony/config and symfony/dependency-injection. Agreed on not tying drupal/core to the beta release; now using 8.5.x as suggested.

Tested and it all builds as expected.

**Comment by unclegcb** [comment]
Nice: looks good to me. Circle tests failed applying core patches, though. did some of these got rolled in to 8.5?

**Comment by ruscoe** [comment]
@unclegcb One of them did (removed from composer.json now). I'll double check the others.

**Comment by ruscoe** [comment]
The patched issues are still open and all the patches apply locally. 

```bash
  - Applying patches for drupal/core
    https://www.drupal.org/files/issues/2484693-47.patch (Telephone Link field formatter breaks Drupal with 5 digits or less in the number)
    https://www.drupal.org/files/issues/2567035-surpress-uasort-contents-modified-errors-20.patch (Database::startLog() cause a notice in Shortcut::sort())
    https://www.drupal.org/files/issues/2699157-23.drupal.Plugin-Lazy-loading-can-cause-usort-warning.patch (Plugin Lazy loading can cause usort warning)
```

Patches are failing for drupal/ctools and drupal/entity_browser too. Both valid patches. Could Circle have a problem actually downloading the patches? The error messages aren't super informative.

**Comment by bdimaggio** [comment]
Just want to note that I didn't do anything special to address those problems you were reporting with patches getting applied, @ruscoe , but CircleCI is no longer complaining about them. So... yay?

**Pull request merged**